pennylane/qnodes/base.py

```diff
@@ -477,7 +527,11 @@ def _default_args(self, kwargs):
         # apply defaults
         for name, s in self.func.sig.items():
             default = s.par.default
-            if default != inspect.Parameter.empty:
+            correct_default = all(d != inspect.Parameter.empty for d in default)\
+                              if isinstance(default, np.ndarray)\
+                              else default != inspect.Parameter.empty
+
+            if correct_default:
```

Comment:
@antalszava, were these changes to the logic made to allow a particular set of tests to pass? I'm slightly confused here

Reply:
Exactly. For one of the test cases, a ``np.array`` is passed coming from ``BasisState`` and then ``default != inspect.Parameter.empty`` throws an error.